### PR TITLE
Add v2 migration logic for prometheus retention

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

Add v2 migration logic for prometheus retention (`prometheus-operator.prometheus.prometheusSpec.retention`)

---

###### Testing performed

- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Review by astencel-sumo — DISMISSED** [review]

File: deploy/helm/sumologic/upgrade-2.0.0.sh [hunk 1]
```diff
@@ -177,6 +177,25 @@ function migrate_prometheus_operator_to_kube_prometheus_stack() {
...
+  readonly RETENTION="$(yq r "${TEMP_FILE}" -- 'prometheus-operator.prometheus.prometheusSpec.retention')"
+
+  if [[ -z "${RETENTION}" ]]; then
+    return
+  fi
+
+  if [[ "${RETENTION}" == "7d" ]]; then
+    info "Changing prometheus retention period from 7d to 1d"
+  elif [[ "${RETENTION}" != "1d" ]]; then
+    warning "Changing prometheus retention period from (custom set) ${RETENTION} to 1d"
```
Comment (perk-sumo) [inline deploy/helm/sumologic/upgrade-2.0.0.sh:194]: I don't think we should change it in such case.

**Review by perk-sumo — CHANGES_REQUESTED** [review]

File: deploy/helm/sumologic/upgrade-2.0.0.sh (hunk 1 above)
Comment (perk-sumo) [reply deploy/helm/sumologic/upgrade-2.0.0.sh:194]: I mean - if someone changed that from the default 7d they probably knew what they were doing.
I think we should rather inform user about the change of the default value for the chart and explicitly say that we leave it as it is.

Comment (pmalek-sumo) [reply deploy/helm/sumologic/upgrade-2.0.0.sh]: Changed, please recheck.

**Review by astencel-sumo — APPROVED** [review]

**Review by perk-sumo — APPROVED** [review]
👍

File: deploy/helm/sumologic/upgrade-2.0.0.sh (hunk 1 above)
Comment (perk-sumo) [reply deploy/helm/sumologic/upgrade-2.0.0.sh]: Thanks :)

**Pull request merged**